File: src/services/performance-history/functions/KAR_HST_add_performance.ts
import { EventBridgeEvent, EventBridgeHandler } from 'aws-lambda';
import performanceService from '../database';
import Performance from '../domain/Performance';

export const handler = async (event: EventBridgeEvent<string, Performance>) => {
  try {
    const performance = event.detail;
    console.log(performance);
    const result = await performanceService.createPerformance(performance);
    console.log(result);

    return {
      statusCode: 201,
      result,
    };
  } catch (err) {
    return {
      statusCode: 500,
      error: err,
    };
  }
};

File: src/karaoke/performance_history/KAR_HST_add_performance.py
import sys
import logging
import traceback
import json
from performances_db import create_performance

logger = logging.getLogger()
logger.setLevel(logging.INFO)

def handler(event, context):
    performance = event['detail']

    try:
        create_performance(performance)
    except Exception as e:
        exception_type, exception_value, exception_traceback = sys.exc_info()
        traceback_string = traceback.format_exception(exception_type, exception_value, exception_traceback)
        err_msg = json.dumps({
            "errorType": exception_type.__name__,
            "errorMessage": str(exception_value),
            "stackTrace": traceback_string
        })
        logger.error(err_msg)